Source/WebCore/bindings/js/JSEventTargetCustom.h

```diff
@@ -47,6 +47,8 @@ class JSEventTargetWrapper {
 
     operator JSC::JSObject&() { ASSERT(m_wrapper); return *m_wrapper; }
 
+    JSDOMGlobalObject* globalObject() const { ASSERT(m_wrapper); return JSC::jsCast<JSDOMGlobalObject*>(m_wrapper->globalObject()); }
```

Comment:
The CodeGenerator wanted to call a `*castedThis->globalObject()` which produced a "member not defined" error.

I am unsure if this is "right", but it got the code to compile, and tests to pass. 

My question is, should the `globalObject` be resolved form the `m_wrapped` instead?

Reply:
For sake of visibility; these are the impacts on the generated code:

```cpp
// WebKitBuild/Debug/DerivedSources/WebCore/JSEventTarget.cpp
void JSEventTargetPrototype::finishCreation(VM& vm)
{
    Base::finishCreation(vm);
    reifyStaticProperties(vm, JSEventTarget::info(), JSEventTargetPrototypeTableValues, *this);
    bool hasDisabledRuntimeProperties = false;
    if (!(((globalObject())->inherits<JSDOMWindowBase>() || (globalObject())->inherits<JSWorkerGlobalScopeBase>()) && jsCast<JSDOMGlobalObject*>(globalObject())->scriptExecutionContext()->settingsValues().observableEnabled)) {
        hasDisabledRuntimeProperties = true;
        auto propertyName = Identifier::fromString(vm, "when"_s);
        VM::DeletePropertyModeScope scope(vm, VM::DeletePropertyMode::IgnoreConfigurable);
        DeletePropertySlot slot;
        JSObject::deleteProperty(this, globalObject(), propertyName, slot);
    }
    if (hasDisabledRuntimeProperties && structure()->isDictionary())
        flattenDictionaryObject(vm);
    JSC_TO_STRING_TAG_WITHOUT_TRANSITION();
}

static inline JSC::EncodedJSValue jsEventTargetPrototypeFunction_whenBody(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSEventTarget>::ClassParameter castedThis)
{
    SUPPRESS_UNCOUNTED_LOCAL auto& vm = JSC::getVM(lexicalGlobalObject);
    auto throwScope = DECLARE_THROW_SCOPE(vm);
    UNUSED_PARAM(throwScope);
    UNUSED_PARAM(callFrame);
    SUPPRESS_UNCOUNTED_LOCAL auto& impl = castedThis->wrapped();
    if (UNLIKELY(callFrame->argumentCount() < 1))
        return throwVMError(lexicalGlobalObject, throwScope, createNotEnoughArgumentsError(lexicalGlobalObject));
    auto* context = jsCast<JSDOMGlobalObject*>(lexicalGlobalObject)->scriptExecutionContext();
    if (UNLIKELY(!context))
        return JSValue::encode(jsUndefined());
    EnsureStillAliveScope argument0 = callFrame->uncheckedArgument(0);
    auto eventTypeConversionResult = convert<IDLAtomStringAdaptor<IDLDOMString>>(*lexicalGlobalObject, argument0.value());
    if (UNLIKELY(eventTypeConversionResult.hasException(throwScope)))
       return encodedJSValue();
    EnsureStillAliveScope argument1 = callFrame->argument(1);
    auto optionsConversionResult = convert<IDLDictionary<ObservableEventListenerOptions>>(*lexicalGlobalObject, argument1.value());
    if (UNLIKELY(optionsConversionResult.hasException(throwScope)))
       return encodedJSValue();
// NOTE: 🚨 IT WAS THIS *castedThis->globalObject() THAT WAS CAUSING COMPILER ERRORS
    RELEASE_AND_RETURN(throwScope, JSValue::encode(toJS<IDLInterface<Observable>>(*lexicalGlobalObject, *castedThis->globalObject(), throwScope, WebCore::Observable::when(impl, *context, eventTypeConversionResult.releaseReturnValue(), optionsConversionResult.releaseReturnValue()))));
}
```